lib/Runtime/Library/JavascriptProxy.h

```diff
@@ -52,6 +52,7 @@ namespace Js
         JavascriptProxy(DynamicType * type, ScriptContext * scriptContext, RecyclableObject* target, RecyclableObject* handler);
         static BOOL Is(_In_ Var obj);
         static BOOL Is(_In_ RecyclableObject* obj);
+        BOOL IsRevoked();
```

Comment:
nit: I personally think all new APIs should be `bool` rather than `BOOL`

Reply:
I was just following what was done in the file, changed now though.

Reply:
I don’t understand the motivation for returning `const bool`?  The `const` in this context seems redundant since `bool` is an immutable value type.  What am I missing?

Reply:
@fatcerberus I meant more like `bool IsRevoked() const;`, since it can be called on const proxies with no side effects.

Reply:
oops fixed (and squashed)

Reply:
Also for clarification on the BOOL vs bool, BOOL was the old (microsoft-specific?) version before `bool` was part of the C++ standard (and I think its still not even in C, ridiculously). Its main problem is that its typedef'd to `int`, which causes silly conversion warnings among other things. Old APIs will remain BOOL indefinitely since changing it would affect almost every file (not to mention some uses of BOOL are for interaction with Win32 APIs that expect it), but I think we have been generally trying to avoid it for anything new unless it needs to interop.

Reply:
C does have bool now, but you have to include stdbool.h for the typedef as it’s normally called `_Bool`.  I use it in my own code, it was a long overdue addition.